hashes/src/main/kotlin/org/kethereum/hashes/ArrayExtensions.kt

```diff
@@ -0,0 +1,100 @@
+package org.kethereum.hashes
+
+// INT ARRAY
+
+/**
+ * Converts the given int array into a byte array via big-endian conversion
+ * (1 int becomes 4 bytes).
+ * @return The converted array.
+ */
+fun IntArray.toByteArray(): ByteArray {
```

Comment:
Or we move these functions to another module if they could be helpful there. But exposing them from hashes feels wrong. Also think we should move the sha256 implementation to a separate module. Hashes was just intended to be a proxy to easily switch away from the java implementation ;-)

Reply:
Ok so I've made them all `internal`, but I believe that all the SHA256 implementation should be moved somewhere else, maybe inside a new module called with a proper name such as `crypto` or something like that. You are free to do it as IDK how you want to call it.  

BTW, I've seen that #35 has been merged, but inside the `hash` module the BouncyCastle implementation is still used instead of the pure-Kotlin one.